students/steve-long/lesson05-excepts-comps/ex-5-4-mailroom_part_3/mailroom3.py

```diff
@@ -0,0 +1,1522 @@
+#       several more loops were substituted where the chief result(s) was a
+#       (were) side-effect(s).
+#
+#   These and several clean-up changes are marked in the code under change 002.
+#   Checked with linter flake8.
+#
+# Script Usage:
+# =============
+#
+#   ./mailroom3.py
+#
+# Issues:
+# =======
+#
+#   Application still does not have requirement to save and retrieve NEW
+#   donors. The initial set of donors is hard-coded.
+#
+# History:
+# ========
+#
+# 002/2020-11-02/sal/
+#   Created mailroom v3 from v2 (mailroom3.py).
+# 001/2020-10-25/sal/
+#   Created mailroom v2 from v1 (mailroom2.py). Removed requirements notes for
+#   previous version.
+# 000/2020-10-20/sal/
+#   Created mailroom v1 (mailroom.py).
+#
+# =============================================================================
+
+from functools import reduce
+from datetime import datetime
+import pathlib
+
+# -- Application constants ---------------------------------------------------
+
+
+_BUSINESS_NAME = "ACME Charities"
+_APP_TITLE = "Mailroom"
+_APP_VERSION = "3.0"
+_APP_WIDTH = 100
+
+
+# -- Root data container -----------------------------------------------------
+
+
+_Donors = {}
+
+
+# -- Str utils ---------------------------------------------------------------
+
+def safe_input(prompt):
+    """
+    Return user input, returning None for EOFError or KeyboardInterrupt.
+
+    HIST: 002/New
+    """
+    result = None
+    try:
+        result = input(prompt)
+    except (EOFError, KeyboardInterrupt):
+        result = None
+    return result
+
+
+def clean_name(name):
+    """
+    Remove extraneous white space.
+    HIST: 001/Incorporated Natasha's brevity suggestion.
+          000/New
+    """
+    return " ".join(name.split())
+
+
+def check_name_case(name, override_correction=True):
+    """
+    Clean and capitalize a name. If adjusted name is different from
+    input, optionally allow the user choose preferred spelling. This
+    will account for mixed-case names (Irish, Scot, Klingon, whatever.)
+
+    HIST: 002/Replaced call to "input" with "safe_input".
+          001/Fixed return value for override_correction=False.
+          000/New
+    """
+    revised_name = " ".join(map(lambda nc: nc.capitalize(), name.split()))
+    if (override_correction):
+        if (name != revised_name):
+            answer\
+                = safe_input("Use \"{}\" [Y] or original spelling [N]?: "
+                             .format(revised_name))
+            if (answer[0:1].upper() != "Y"):
+                print("Reverting to original spelling")
+                revised_name = name
+    return revised_name
+
+
+def is_floatable(v):
+    """
+    Can a value be coerced to a floating-point value? Implemented
+    because just checking for <v>.isnumeric() won't work with a value
+    containing a decimal.
+
+    HIST: 001/Incorporated Natasha's brevity suggestion.
+          000/New
+    """
+    try:
+        float(v)
+    except Exception:
```

Comment:
note that it is best practice to catch specific exceptions like TypeError/ValueError etc

it is pretty obvious in this case though on what you're attempting to catch.